### PR TITLE
[Doc] Add custom dataset tutorial

### PR DESCRIPTION
**Pull request by tonysy**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: docs/en/user_guides/4_pretrain_custom_dataset.md [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+
+In this tutorial, we provide some tips on how to conduct self-supervised learning on your own dataset(without the need of label).
+
+## Train MAE on Custom Dataset
+
+In MMSelfSup, We support the `CustomDataset` from MMClassification,  which is able to read the images within the specified folder directly. You only need to prepare the path information of the custom dataset and edit the config.
+
+### Get the path of custom dataset
+
+It should be like `/dataset/my_custom_dataset`
```
Comment (fangyixiao18) [inline docs/en/user_guides/4_pretrain_custom_dataset.md]: Better to keep consistent with config (`'data/custom_dataset/'`)

File: docs/en/user_guides/4_pretrain_custom_dataset.md [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+randomness = dict(seed=0, diff_rank_seed=True)
+resume = True
+```
+
+### Edit the dataset related config
+
+The dataset related config is defined in `'../_base_/datasets/imagenet_mae.py'` in `_base_`. We then copy the content of dataset config file into our created file `mae_vit-base-p16_8xb512-coslr-400e_custom_dataset.py`.
+
+- Then we remove the `'../_base_/datasets/imagenet_mae.py'` in `_base_`.
+- Set the `dataset_type = 'mmcls.CustomDataset'`, and the path of the custom dataset ` dataset_type = /dataset/my_custom_dataset`.
```
Comment (fangyixiao18) [inline docs/en/user_guides/4_pretrain_custom_dataset.md]:
```suggestion
- Set the `dataset_type = 'mmcls.CustomDataset'`, and the path of the custom dataset ` data_root = data/custom_dataset/`.
```

File: docs/en/user_guides/4_pretrain_custom_dataset.md [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+
+# randomness
+randomness = dict(seed=0, diff_rank_seed=True)
+resume = True
+```
+
+By using the edited config file, you are able to train a self-supervised model with MAE algorithm on the custom dataset.
+
+## Train MAE on COCO Dataset
+
```
Comment (fangyixiao18) [inline docs/en/user_guides/4_pretrain_custom_dataset.md:190]: Better to remind users to install MMDet and check the version at the beginning

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: docs/en/user_guides/4_pretrain_custom_dataset.md [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+    checkpoint=dict(type='CheckpointHook', interval=10, max_keep_ckpts=3))
+
+```
+
+## Load pre-trained model to speedup convergence
+
+To speedup the convergence of the model on your own dataset. You may use the pre-trained model as the initialization for the model's weight. You just need to specify the url of the pre-trained model via command.
+
+```bash
+bash tools/dist_train.sh ${CONFIG} ${GPUS} --cfg-options ${PRETRAIN}
```
Comment (fangyixiao18) [inline docs/en/user_guides/4_pretrain_custom_dataset.md]: --cfg-options model.pretrained=......

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]

File: docs/en/user_guides/4_pretrain_custom_dataset.md [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+    - [Edit the dataset related config](#edit-the-dataset-related-config)
+  - [Train MAE on COCO Dataset](#train-mae-on-coco-dataset)
+  - [Train SimCLR on Custom Dataset](#train-simclr-on-custom-dataset)
+  - [Load Pre-trained Model to Speedup Convergence](#load-pre-trained-model-to-speedup-convergence)
+
+In this tutorial, we provide some tips on how to conduct self-supervised learning on your own dataset(without the need of label).
+
+## Train MAE on Custom Dataset
+
+In MMSelfSup, We support the `CustomDataset` from MMClassification,  which is able to read the images within the specified folder directly. You only need to prepare the path information of the custom dataset and edit the config.
```
Comment (YuanLiuuuuuu) [inline docs/en/user_guides/4_pretrain_custom_dataset.md]: Maybe we can add something like: similar to the ``ImageFolder`` in ``torchvision``.

File: docs/en/user_guides/4_pretrain_custom_dataset.md [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+
+### Get the path of custom dataset
+
+It should be like `/dataset/my_custom_dataset`
+
+### Choose one config as template
+
+Here, we would like to use `configs/selfsup/mae/mae_vit-base-p16_8xb512-coslr-400e_in1k.py` as the example. We first copy this config file and rename it as `mae_vit-base-p16_8xb512-coslr-400e_custom_dataset.py`.
```
Comment (YuanLiuuuuuu) [inline docs/en/user_guides/4_pretrain_custom_dataset.md]: According to naming rules for config files, `mae_vit-base-p16_8xb512-coslr-400e_custom_dataset.py` should be renamed to `mae_vit-base-p16_8xb512-coslr-400e-${custom_dataset}.py`.

File: docs/en/user_guides/4_pretrain_custom_dataset.md (hunk 2 above)
Comment (YuanLiuuuuuu) [inline docs/en/user_guides/4_pretrain_custom_dataset.md]: Better to remind the user to add the ``custom_dataset`` in ``mmcls``, instead of ``mmselfsup``.

File: docs/en/user_guides/4_pretrain_custom_dataset.md [hunk 7]
```diff
@@ -0,0 +1,370 @@
...
+
+### Edit the dataset related config
+
+The dataset related config is defined in `'../_base_/datasets/imagenet_mae.py'` in `_base_`. We then copy the content of dataset config file into our created file `mae_vit-base-p16_8xb512-coslr-400e_custom_dataset.py`.
+
+- Then we remove the `'../_base_/datasets/imagenet_mae.py'` in `_base_`.
+- Set the `dataset_type = 'mmcls.CustomDataset'`, and the path of the custom dataset ` dataset_type = /dataset/my_custom_dataset`.
+- Remove the `ann_file` in `train_dataloader`, and edit the `data_prefix` if needed.
+
+And edited the config will be like this:
```
Comment (YuanLiuuuuuu) [inline docs/en/user_guides/4_pretrain_custom_dataset.md]:
```suggestion
And the edited config will be like this:
```

File: docs/en/user_guides/4_pretrain_custom_dataset.md [hunk 8]
```diff
@@ -0,0 +1,370 @@
...
+    checkpoint=dict(type='CheckpointHook', interval=1, max_keep_ckpts=3))
+
+# randomness
+randomness = dict(seed=0, diff_rank_seed=True)
+resume = True
+```
+
+## Train SimCLR on Custom Dataset
+
+We provide an example of using SimCLR on custom dataset, the main idea is similar with the [Train MAE on Custom Dataset
```
Comment (YuanLiuuuuuu) [inline docs/en/user_guides/4_pretrain_custom_dataset.md]:
```suggestion
We provide an example of using SimCLR on custom dataset, the main idea is similar to the [Train MAE on Custom Dataset
```

**Review by YuanLiuuuuuu — APPROVED** [review]

**Review by fangyixiao18 — APPROVED** [review]
